lib/trino-parquet/src/test/java/io/trino/parquet/reader/BenchmarkShortDecimalColumnReader.java

```diff
@@ -0,0 +1,210 @@
+import static java.lang.Math.toIntExact;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.apache.parquet.hadoop.metadata.CompressionCodecName.UNCOMPRESSED;
+import static org.apache.parquet.schema.PrimitiveType.PrimitiveTypeName.FIXED_LEN_BYTE_ARRAY;
+import static org.joda.time.DateTimeZone.UTC;
+
+@State(Scope.Thread)
+@OutputTimeUnit(SECONDS)
+@Measurement(iterations = 15, time = 500, timeUnit = MILLISECONDS)
+@Warmup(iterations = 10, time = 500, timeUnit = MILLISECONDS)
+@Fork(3)
+@SuppressWarnings("SameParameterValue")
+public class BenchmarkShortDecimalColumnReader
+{
+    // Parquet pages are usually about 1MB
+    private static final int MIN_PAGE_SIZE = 1_000_000;
+    private static final int OUTPUT_BUFFER_SIZE = MIN_PAGE_SIZE * 2; // Needs to be more than MIN_PAGE_SIZE
+    private static final int MAX_VALUES = 5_000_000;
+
+    private static final int DATA_GENERATION_BATCH_SIZE = 16384;
+    private static final int READ_BATCH_SIZE = 4096;
+
+    private final List<DataPage> dataPages = new ArrayList<>();
+
+    @Param({
+            "1", "2", "3", "4", "5", "6", "7", "8",
+    })
+    private int byteArrayLength;
+
+    public BenchmarkShortDecimalColumnReader()
+    {
+    }
+
+    public BenchmarkShortDecimalColumnReader(int byteArrayLength)
+    {
+        this.byteArrayLength = byteArrayLength;
+    }
+
+    @Setup
+    public void setup()
+            throws IOException
+    {
+        ValuesWriter writer = createValuesWriter(OUTPUT_BUFFER_SIZE);
+        int precision = maxPrecision(byteArrayLength);
+        int positions = 0;
+        int batchIndex = 0;
+        long[] batch = generateDataBatch(DATA_GENERATION_BATCH_SIZE, precision);
+        while (writer.getBufferedSize() < MIN_PAGE_SIZE && positions < MAX_VALUES) {
+            if (batchIndex == DATA_GENERATION_BATCH_SIZE) {
+                dataPages.add(createDataPage(writer, batchIndex));
+                batch = generateDataBatch(DATA_GENERATION_BATCH_SIZE, precision);
+                batchIndex = 0;
+            }
+            writeValue(writer, batch, batchIndex++);
+            positions++;
+        }
+
+        if (batchIndex > 0) {
+            dataPages.add(createDataPage(writer, batchIndex));
+        }
+    }
+
+    @Benchmark
+    public void read()
+            throws IOException
+    {
+        PrimitiveField field = getField();
+        PrimitiveColumnReader columnReader = PrimitiveColumnReader.createReader(field, UTC);
+        columnReader.setPageReader(new PageReader(UNCOMPRESSED, new LinkedList<>(dataPages), null, MAX_VALUES), null);
+        int rowsRead = 0;
+        while (rowsRead < MAX_VALUES) {
+            int remaining = MAX_VALUES - rowsRead;
+            columnReader.prepareNextRead(Math.min(READ_BATCH_SIZE, remaining));
+            rowsRead += columnReader.readPrimitive(field).getBlock().getPositionCount();
```

Comment:
You need to do something with the results. Otherwise, the VM might detect some code as dead code and optimize it out. Return `rowsRead` from this method